### PR TITLE
Upload api multipart upload

### PR DESCRIPTION
**Pull request by SteveRedka**

### CODE REVIEW

File: lib/uploadcare/client/chunks_client.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Uploadcare
+  # This class splits file into chunks of set chunk_size
+  # and uploads them into cloud storage.
+  # Used for multipart uploads
+  # https://uploadcare.com/api-refs/upload-api/#tag/Upload/paths/https:~1~1uploadcare.s3-accelerate.amazonaws.com~1%3C%3Cpresigned-url%3E/put
+  class ChunksClient < ApiStruct::Client
```
Comment (dmitry-mukhin) [inline lib/uploadcare/client/chunks_client.rb]: we don't need this class. the logic should be part of multipartuploadclient

File: lib/uploadcare/uploadcare_settings.rb [hunk 2]
```diff
@@ -4,4 +4,5 @@ module Uploadcare
   PUBLIC_KEY = ENV.fetch('UPLOADCARE_PUBLIC_KEY')
   SECRET_KEY = ENV.fetch('UPLOADCARE_SECRET_KEY')
   AUTH_TYPE = 'Uploadcare.Simple'
+  MULTIPART_SIZE_THRESHOLD = 10 * 1024 * 1024
```
Comment (dmitry-mukhin) [inline lib/uploadcare/uploadcare_settings.rb]: I believe that this should be 100 MB, not 10 MB by default.
```suggestion
  MULTIPART_SIZE_THRESHOLD = 100 * 1024 * 1024
```

File: lib/uploadcare/client/chunks_client.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    chunks_api
+
+    def upload_chunks(object, links)
+      chunk_size = 5242880
+      links.each do |link|
+        chunk = object.read(chunk_size)
+        upload_chunk(chunk, link)
+      end
```
Comment (dmitry-mukhin) [inline lib/uploadcare/client/chunks_client.rb]: how easy/difficult would be to do this in parallel via async io or threads?

File: lib/uploadcare/client/chunks_client.rb (hunk 1 above)
Comment (dmitry-mukhin) [reply lib/uploadcare/client/chunks_client.rb]: I see that this comes from ApiStruct API.
But can we override route? or, make the class in multipartupload's namespace? it shouldn't be usable on its own.

File: lib/uploadcare/api_struct_settings.rb [hunk 4]
```diff
@@ -14,6 +14,9 @@
     },
     upload_api: {
       root: 'https://upload.uploadcare.com'
+    },
+    chunks_api: {
+      root: ''
```
Comment (dmitry-mukhin) [inline lib/uploadcare/api_struct_settings.rb]: Can these be overridden via settings?
(please, make separate PR if not)

File: lib/uploadcare/client/chunks_client.rb (hunk 1 above)
Comment (SteveRedka) [reply lib/uploadcare/client/chunks_client.rb]: It's a proxy between Amazon API and Uploadcare API. These are two entirely different things

I'll move it into multiupload's namespace

File: lib/uploadcare/uploadcare_settings.rb (hunk 2 above)
Comment (SteveRedka) [reply lib/uploadcare/uploadcare_settings.rb]: 10 MB is minumum size, 100 is maximum for regular uploads

https://uploadcare.com/docs/api_reference/upload/request_based/

File: lib/uploadcare/client/chunks_client.rb (hunk 3 above)
Comment (SteveRedka) [reply lib/uploadcare/client/chunks_client.rb]: Relatively easy. I can make it in another PR

File: lib/uploadcare/uploadcare_settings.rb (hunk 2 above)
Comment (dmitry-mukhin) [reply lib/uploadcare/uploadcare_settings.rb]: minimum does not mean that we should use it.
please set it to 100 MB with a way to override

File: lib/uploadcare/api_struct_settings.rb (hunk 4 above)
Comment (SteveRedka) [reply lib/uploadcare/api_struct_settings.rb]: Technically, it is a settings file which can be replaced by a user

If it is expected that someone will replace API endpoints, some of those settings may be moved into plugin settings

Comment (dmitry-mukhin) [reply lib/uploadcare/api_struct_settings.rb]: yes, it is a requirement to be able to customize these.
at least for CDN and Upload API hosts
